### ISSUE TITLE
Upgrading from 2014062500 to 2017021500 does not work

### ISSUE THREAD

**Issue by abias** [Wunderbyte-GmbH/moodle-mod_booking#77]
Hi David,

I am trying to upgrade your plugin from 2014062500 (on Moodle 2.7) to 2017021500 (on Moodle 3.1). This is a big jump, but it should work.

However, I get this error during the DB upgrade process:

Cannot downgrade mod_booking from 2014062500 to 2014033100.

Debug info: 
Error code: cannotdowngrade

Stack trace:
line 262 of /lib/upgradelib.php: downgrade_exception thrown
line 454 of /mod/booking/db/upgrade.php: call to upgrade_mod_savepoint()
line 726 of /lib/upgradelib.php: call to xmldb_booking_upgrade()
line 432 of /lib/upgradelib.php: call to upgrade_plugins_modules()
line 1742 of /lib/upgradelib.php: call to upgrade_plugins()
line 683 of /admin/index.php: call to upgrade_noncore()

I had a look at db/upgrade.php.
There is a section:
```
    if ($oldversion < 2014033100 || $oldversion < 2014092901) {
[...]
        // Booking savepoint reached.
        upgrade_mod_savepoint(true, 2014033100, 'booking');
    }

```

My old version 2014062500 does match the second criteria in the if clause and upgrade_mod_savepoint() tries then to downgrade the plugin which fails.

Could you please double-check what goes wrong here?

Thanks,
Alex

**Comment by dasistwas** [comment]
Hi,

thanks for reporting this problem.  This is now solved in a53839a9bd5f2bd2c18a3734f17d3a4d1b5f7e1f

Kind regards,
David

**Comment by abias** [comment]
Thanks, David, for the quick fix.
